### ISSUE TITLE
Grid::new fails for planar meshes due to zero volume

### ISSUE THREAD

**Issue by Nevsden** [dima634/baby_shark#36]
For planar meshes, incremental remeshing cannot be used, as no grid can be built for this type of meshes.
I produced a dirty hack to fix the zero volume issue for planar meshes. The incremental remeshing then works for planar meshes. The code below shows this hack. The hack does only work for planar meshes with a z-vector normal. For an x-vector and y-vector normal, one would need a separate hack.

This cannot be the final solution, I guess.

```rust
/// Compute a reasonable size for the cell such that
    /// the number of cells is the same of the numbers of elements to be inserted in the grid
    fn calculate_cell_size(objects: &Vec<TObject>) -> Vector3<TObject::ScalarType> {
        // Compute bbox of all objects
        let mut bbox = objects[0].bbox();
        for object in objects {
            bbox.union_box(&object.bbox());
        }
        let number_of_cells = objects.len() as f64;
        let cell_size: TObject::ScalarType = if bbox.size_z() == Zero::zero() {
            let cell_area =
                cast::<TObject::ScalarType, f64>(bbox.area()).unwrap() / number_of_cells;
            cast(cell_area.sqrt()).unwrap()
        } else {
            let cell_volume =
                cast::<TObject::ScalarType, f64>(bbox.volume()).unwrap() / number_of_cells;
            cast(cell_volume.cbrt()).unwrap()
        };

        Vector3::new(cell_size, cell_size, cell_size)
    }
```

**Comment by dima634** [comment]
It has been fixed by this [commit ](https://github.com/dima634/baby_shark/commit/98de5efe4678890e1bffdb719fd5c9958a046541)and is included in the crate version `0.3.3`. Please let me know if I can close this issue.

**Comment by Nevsden** [comment]
Looks good to me. I like this solution, as it is independent of the direction of the planar mesh. Great.
